### PR TITLE
Provide synchronizable create_parameter in PARL

### PR DESCRIPTION
**Pull request by zenghsh3**
- provide synchronizable create_parameter api
- add corresponding unittest

### CODE REVIEW

**Comment by zenghsh3** [comment]
- use AttrHold to make LayerFunc support more than two parameters, fix #25

File: parl/layers/layer_wrappers.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 import paddle.fluid.unique_name as unique_name
 import paddle.fluid as fluid
 import six
+from .utils import AttrHolder
```
Comment (TomorrowIsAnOtherDay) [inline parl/layers/layer_wrappers.py]: To keep this line consistent with other codes that describes the code path, use absolute path here, e.g., parl.layers.utils

File: parl/layers/layer_wrappers.py [hunk 2]
```diff
@@ -91,15 +92,15 @@ def create_new_para_name(attr):
...
             return None
 
     @property
     def bias_name(self):
-        if self.bias_attr:
-            return self.bias_attr.name
+        if self.attr_holder.bias_attr:
+            return self.attr_holder.bias_attr.name
         else:
             return None
```
Comment (TomorrowIsAnOtherDay) [inline parl/layers/layer_wrappers.py:106]: Except properties of `param_name` and `bias_name` above, users can't acquire other parameters' name, how about adding a new property for users to get all the parameters_name ?

File: parl/layers/utils.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# limitations under the License.
+
+import six
+from copy import deepcopy
+
+__all__ = ['AttrHolder']
+
+
+class AttrHolder(object):
+    """ Maintain a dict of attrs """
```
Comment (TomorrowIsAnOtherDay) [inline parl/layers/utils.py]: Please describe the occasions where to use this class.

File: parl/plutils/common.py [hunk 4]
```diff
@@ -82,11 +78,10 @@ def get_parameter_pairs(src, target):
             target_var = getattr(target, attr)
             param_pairs.extend(get_parameter_pairs(src_var, target_var))
     elif isinstance(src, LayerFunc):
-        param_pairs.append((src.param_attr.name, target.param_attr.name,
-                            False))
-        if src.bias_attr:
-            param_pairs.append((src.bias_attr.name, target.bias_attr.name,
-                                True))
+        for (src_attr, target_attr) in zip(src.attr_holder.sorted(),
```
Comment (TomorrowIsAnOtherDay) [inline parl/plutils/common.py]: How can you guarantee that `src.attr_holder` and `target.target_holder` have same number of parameters here?

File: parl/layers/layer_wrappers.py (hunk 1 above)
Comment (zenghsh3) [reply parl/layers/layer_wrappers.py]: done

File: parl/layers/layer_wrappers.py (hunk 2 above)
Comment (zenghsh3) [reply parl/layers/layer_wrappers.py:106]: done

File: parl/layers/utils.py (hunk 3 above)
Comment (zenghsh3) [reply parl/layers/utils.py]: done

File: parl/plutils/common.py (hunk 4 above)
Comment (zenghsh3) [reply parl/plutils/common.py]: added assertion here.

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
LGTM.

**Pull request merged**